crates/apollo-compiler/src/validation/selection.rs

```diff
@@ -67,13 +67,28 @@ impl<'a> FieldSelection<'a> {
 pub(crate) fn expand_selections<'doc>(
     fragments: &'doc IndexMap<Name, Node<executable::Fragment>>,
     selection_sets: impl Iterator<Item = &'doc executable::SelectionSet>,
+    mut for_subscription_top_level: Option<(&executable::Operation, &mut DiagnosticList)>,
 ) -> Vec<FieldSelection<'doc>> {
     let mut selections = vec![];
     let mut queue: VecDeque<&executable::SelectionSet> = selection_sets.collect();
     let mut seen_fragments = HashSet::with_hasher(ahash::RandomState::new());
 
     while let Some(next_set) = queue.pop_front() {
         for selection in &next_set.selections {
+            if let Some((operation, diagnostics)) = &mut for_subscription_top_level {
+                if let Some(conditional_directive) = selection
+                    .directives()
+                    .iter()
+                    .find(|d| matches!(d.name.as_str(), "skip" | "include"))
+                {
+                    diagnostics.push(
+                        conditional_directive.location(),
+                        executable::BuildError::SubscriptionUsesConditionalSelection {
+                            name: operation.name.clone(),
+                        },
+                    );
+                }
+            }
```

Comment:
I don't think this should be part of `expand_selections`. That function is for expanding selections, not for doing validation. We can just check this in `validate_subscription` with the list returned by `expand_selections`